### ISSUE TITLE
tuple index out of range

### ISSUE THREAD

**Issue by 979736316** [lululxvi/deepxde#408]
Dear Dr.Lu
    I am trying to use Deepxde for PDE discovery, I set up many partial differential operators and want to learn the coefficients in front of the operators through the data, but unfortunately, it shows an error when there are too many coefficients, what should I do?
![image](https://user-images.githubusercontent.com/52651366/139177707-5379d0d3-709c-40bd-b802-5d50d619ecd7.png)

- For example, I set the differential operator u, u**2, u**3, u**2*uxx，….The corresponding previous coefficients are lambda1,lambda2,lambda3,lambda4,…
- How to apply the l0 norm to all lambda, i.e. the loss value becomes of the following form
![image](https://user-images.githubusercontent.com/52651366/139178355-2acf244d-469f-4808-897c-745dccff8888.png)

**Comment by 979736316** [comment]
Thank you for your reply, I've solved the problem. Now it's working properly. Should I apply the L0 norm to all coefficients?

**Comment by lululxvi** [comment]
DeepXDE current only has L1/L2 norm. You may check if TensorFlow or PyTorch has L0 norm.
